src/perform_step/rosenbrock_perform_step.jl

```diff
@@ -37,9 +37,15 @@ end
   dto2 = dt/2
   dto6 = dt/6
 
-  calc_rosenbrock_differentiation!(integrator, cache, γ, γ, repeat_step, false)
+  new_W = calc_rosenbrock_differentiation!(integrator, cache, γ, γ, repeat_step, false)
+
+  calculate_residuals!(weight, fill!(weight, one(eltype(u))), uprev, uprev,
+                       opts.abstol, opts.reltol, opts.internalnorm, t)
+
+  cache.linsolve(vec(k₁), W, vec(linsolve_tmp), new_W,
+      Pl=DiffEqBase.ScaleVector(weight, true),
+      Pr=DiffEqBase.ScaleVector(weight, false), reltol=opts.reltol)
 
-  cache.linsolve(vec(k₁), W, vec(linsolve_tmp), !repeat_step)
```

Comment:
do all of the Rosenbrocks need this treatment?

Reply:
Yes